storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -3108,10 +3108,1655 @@ buf_block_t *fsp_get_latched_xdes_page(
+{
+  if (!flst_get_len(base->page.frame + boffset))
+    return DB_SUCCESS;
+
+  fil_addr_t addr= flst_get_last(base->page.frame + boffset);
+
+  if (addr.page >= free_limit)
+    return DB_CORRUPTION;
+
+  if (!flst_node_offset_validate(&addr))
+    return DB_CORRUPTION;
+
+  dberr_t err= DB_SUCCESS;
+  *last_block_list= fsp_get_latched_page(page_id_t{0, addr.page},
+                                         mtr, &err);
+  return err;
+}
+
+/** Complete the steps for adding the block into last of the list
+@param base       base block where free list starts
+@param boffset    offset where list starts
+@param curr       extent descriptor block
+@param coffset    offset to point the descriptor
+@param mtr        mini-transaction */
+void flst_add_last_complete(buf_block_t *base, uint16_t boffset,
+                            buf_block_t *curr, uint16_t coffset,
+                            mtr_t *mtr) noexcept
+{
+  fil_addr_t last_addr= flst_get_last(base->page.frame + boffset);
+  buf_block_t *last_block_list= nullptr;
+  if (last_addr.page != FIL_NULL)
+  {
+    last_block_list= mtr->get_already_latched(page_id_t{0, last_addr.page},
+                                             MTR_MEMO_PAGE_SX_FIX);
+    ut_ad(last_block_list);
+  }
+
+  if (last_block_list == nullptr)
+  {
+    flst_write_addr(*curr,
+                    curr->page.frame + coffset + FLST_PREV,
+                    FIL_NULL, 0, mtr);
+    flst_write_addr(*base, base->page.frame + boffset + FLST_FIRST,
+                    curr->page.id().page_no(), coffset, mtr);
+    memcpy(base->page.frame + boffset + FLST_LAST,
+           base->page.frame + boffset + FLST_FIRST, FIL_ADDR_SIZE);
+    mtr->memmove(*base, boffset + FLST_LAST,
+		 boffset + FLST_FIRST, FIL_ADDR_SIZE);
+  }
+  else
+  {
+    fil_addr_t addr= flst_get_last(base->page.frame + boffset);
+
+    flst_write_addr(*last_block_list,
+                    last_block_list->page.frame + addr.boffset
+		      + FLST_NEXT,
+                    curr->page.id().page_no(), coffset, mtr);
+    flst_write_addr(*curr,
+                    curr->page.frame + coffset + FLST_PREV,
+                    addr.page, addr.boffset, mtr);
+    flst_write_addr(*base, base->page.frame + boffset + FLST_LAST,
+                    curr->page.id().page_no(), coffset, mtr);
+  }
+
+  flst_write_addr(*curr,
+                  curr->page.frame + coffset + FLST_NEXT,
+                  FIL_NULL, 0, mtr);
+
+  byte *len= base->page.frame + boffset + FLST_LEN;
+  mtr->write<4>(*base, len, mach_read_from_4(len) + 1);
+}
+
+/** Prepare the associate pages of the current block and modify
+the associated pages */
+class associatedPages
+{
+  buf_block_t *m_left_block= nullptr;
+  buf_block_t *m_right_block= nullptr;
+  buf_block_t *m_parent_block= nullptr;
+  buf_block_t *m_cur_block= nullptr;
+  mtr_t *m_mtr;
+
+public:
+  associatedPages(buf_block_t *cur_block, mtr_t *mtr)
+    : m_cur_block(cur_block), m_mtr(mtr) {}
```

Comment:
We don’t seem to need a default initializer for `m_cur_block`, because the only constructor is initializing it.